src/lib/models/ConfigModel.ts

```diff
@@ -176,31 +192,29 @@ abstract class YamlFormatter {
 	}
 
 	private toYamlEngineOverridesForEngine(engineName: string): string {
-		const engineConfigDescriptor = this.userContext.core.getEngineConfigDescription(engineName);
-		const userEngineConfig = this.userContext.core.getEngineConfig(engineName);
-		const defaultEngineConfig = this.defaultContext.core.getEngineConfig(engineName);
+		const engineConfigDescriptor: ConfigDescription = this.codeAnalyzer.getEngineConfigDescription(engineName);
+		const userEngineConfig: EngineConfig = this.codeAnalyzer.getEngineConfig(engineName);
 
 		let yamlCode: string = '\n' +
-			this.toYamlSectionHeadingComment(engineConfigDescriptor.overview!) + '\n' +
+			this.toYamlSectionHeadingComment(engineConfigDescriptor.overview) + '\n' +
 			`${engineName}:\n`;
 		// By fiat, the field description will always include, at minimum, an entry for "disable_engine", so we can
 		// assume that the object is not undefined.
-		for (const configField of Object.keys(engineConfigDescriptor.fieldDescriptions!)) {
-			const fieldDescription = engineConfigDescriptor.fieldDescriptions![configField];
-			const defaultValue = defaultEngineConfig[configField] ?? null;
-			const userValue = userEngineConfig[configField] ?? defaultValue;
+		for (const configField of Object.keys(engineConfigDescriptor.fieldDescriptions)) {
+			const fieldDescription: ConfigFieldDescription = engineConfigDescriptor.fieldDescriptions[configField];
+			const userValue = userEngineConfig[configField] ?? fieldDescription.defaultValue;
 			// Add a leading newline to visually break up the property from the previous one.
 			yamlCode += '\n' +
-				indent(this.toYamlComment(fieldDescription), 2) + '\n' +
-				indent(this.toYamlField(configField, userValue, defaultValue), 2) + '\n';
+				indent(this.toYamlComment(fieldDescription.descriptionText), 2) + '\n' +
+				indent(this.toYamlFieldUsingFieldDescription(configField, userValue, fieldDescription), 2) + '\n';
 		}
 		return yamlCode.trimEnd();
 	}
 }
 
 class PlainYamlFormatter extends YamlFormatter {
```

Comment:
There still is a bit of a code smell having to pass all these input arguments all around. Part of me thinks that we could maybe just give the Formatter the Model instead of passing the values through. But i'm on the fence... so just leaving things as is.